orangecontrib/text/widgets/owcollocations.py

```diff
@@ -0,0 +1,191 @@
+        self.collView = view = TableView(self)  # type: TableView
+        self.mainArea.layout().addWidget(view)
+        view.setModel(model)
+        view.resizeColumnsToContents()
+        view.setItemDelegateForColumn(1, gui.ColoredBarItemDelegate())
+        view.setSelectionMode(QTableView.NoSelection)
+
+    @Inputs.corpus
+    def set_corpus(self, corpus):
+        self.collModel.clear()
+        self.collModel.resetSorting(True)
+        self.corpus = corpus
+        self.commit()
+
+    def _change_type(self):
+        self.type = NGRAM_TYPES[self.type_index]
+        if self.type_index == 1:
+            self.method_rb.buttons[2].setDisabled(True)
+            self.method_rb.buttons[3].setDisabled(True)
+            self.method_rb.buttons[7].setDisabled(True)
+            if self.selected_method in [2, 3, 7]:
+                self.method_rb.buttons[0].click()
+        else:
+            self.method_rb.buttons[2].setDisabled(False)
+            self.method_rb.buttons[3].setDisabled(False)
+            self.method_rb.buttons[7].setDisabled(False)
+        self.commit()
+
+    def compute_scores(self):
+        self.collModel.clear()
+        self.collModel.resetSorting(True)
+        finder = self.type.from_documents(self.corpus.tokens)
+        finder.apply_freq_filter(self.freq_filter)
+
+        res = finder.score_ngrams(self.method.scorer[self.type_index])
+        collocations = np.array([" ".join(col) for col, score in res],
+                                dtype=object)[:, None]
+        scores = np.array([score for col, score in res], dtype=float)[:, None]
+
+        self.results = (collocations, scores)
+        if len(scores):
+            self.collModel.setExtremesFrom(1, scores)
+
+    def commit(self):
+        if self.corpus is None:
+            return
+
+        self.type = NGRAM_TYPES[self.type_index]
+        self.method = SCORING_METHODS[self.selected_method]
+
+        self.compute_scores()
+
+        if not self.results:
+            self.collModel.clear()
+            self.Outputs.corpus.send(None)
+            return
+
+        output = self.create_scores_table()
+        self.collModel[:] = np.hstack(self.results)[:20]
+        self.collView.resizeColumnsToContents()
+
+        self.Outputs.corpus.send(output)
+
+    def create_scores_table(self):
+        col_var, score_var = get_unique_names(self.corpus.domain,
+                                              ["Collocations", "Scores"])
+
+        domain = Domain([ContinuousVariable(col_var)],
+                        metas=[StringVariable(score_var)])
+
+        collocations, scores = self.results
+
+        new_table = Table.from_numpy(domain, scores, metas=collocations)
+        new_table.name = "Collocation Scores"
+        return new_table
+
+    def restore_order(self):
+        """Restore the original data order of the current view."""
+        model = self.collModel
+        if model is not None:
+            model.resetSorting(yes_reset=True)
```

Comment:
```suggestion
        self.collView.horizontalHeader().setSortIndicator(-1, Qt.AscendingOrder)
        if model is not None:
            model.resetSorting(yes_reset=True)
```

Resetting the order does not remove the indicator in the header. You can remove it by adding the line suggested above.